commands.c

```diff
@@ -77,18 +83,28 @@ command_answer *send_command(char *name, char *data, task_context *context)
     spin_unlock_irqrestore(&command_list_lock, command_list_lock_flags);
 
     command_answer *cmd_answer = cmd->answer;
-    if (cmd->context)
-    {
-        free_task_context(cmd->context);
-    }
-    kfree(cmd);
+    free_command(cmd);
 
     return cmd_answer;
 }
 
+// add a command to the list (to be processed by the driver)
+// this is a blocking function, it will wait until the command is processed
+command_answer *send_command(char *name, char *data, task_context *context)
+{
+    return this_send_command(name, data, context, false);
+}
+
+// add a message to the command list (to be processed by the driver)
+// this is a non-blocking function
+command_answer *send_message(char *name, char *data, task_context *context)
```

Comment:
I would change this to `void send_message` to mark it doesn't return anything (now it returns always NULL).